images/hyperkube/Dockerfile.rhel

```diff
@@ -1,6 +1,10 @@
 FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.13 AS builder
 WORKDIR /go/src/github.com/openshift/origin
 COPY . .
+# apply a patch to net/http2 library that fixes the graceful termination of active connections, more details at: https://github.com/golang/go/issues/39776
+# this patch was prepared based on https://go-review.googlesource.com/c/go/+/240278/. The original patch is targeting the master branch and didn't apply to 1.13 version
+RUN patch -d /usr/local/go -p1 < images/hyperkube/4d8de76-1.13.patch
+RUN go test images/hyperkube/gc_test.go
```

Comment:
Would it be better to locate the file in go sources (pkg/ or test/) then put it into images?

Also why does it need to be run separately and not in unit job?

Reply:
My intention was to add it close to the patch. The patch and the tests are temporary.
The patch can be removed when the new version of `go` will be released. Once we have that then I am going to add an integration test for testing graceful shutdown to k/k.

Reply:
IMO having it here makes sense. This is really a smoke test for the build image. It is not part of our product.

Reply:
doesn't it have to be ./images/hyperkube.... ?